MCPClient/Sources/stdioTransport/DataChannel+StdioProcess.swift

```diff
@@ -213,7 +219,9 @@ extension DataChannel {
   private static func loadZshEnvironment() throws -> [String: String] {
     let process = Process()
     process.launchPath = "/bin/zsh"
-    process.arguments = ["-c", "source ~/.zshrc && printenv"]
+    // Those are loaded for interactive login shell by zsh:
+   // https://www.freecodecamp.org/news/how-do-zsh-configuration-files-work/
+    process.arguments = ["-c", "source ~/.zshenv; source ~/.zprofile; source ~/.zshrc; source ~/.zshrc; printenv"]
```

Comment:
nit: below I think looking for `last` instead of `first` is more correct, in the unexpected case sourcing one of those file was also printing the PATH. ie

```swift
if let path = env?.split(separator: "\n").filter({ $0.starts(with: "PATH=") }).last {
```